src/wasm-type.h

```diff
@@ -421,6 +430,8 @@ class HeapType {
   std::string toString() const;
 };
 
+inline bool Type::isNull() const { return isRef() && getHeapType().isBottom(); }
```

Comment:
Unfortunately we can't put the definition in the normal place because it uses `HeapType` by value and therefore has to come after the `HeapType` definition.

We don't want to require any nullability here because most users do not need to differentiate between nullable and non-nullable references to bottom types. For example, in the printer and binary emitter, we need the same behavior when the `ref` child is a reference to a bottom type no matter whether it is nullable or not.

In principle we could optimize even more when we see a non-nullable reference to a bottom type, but we don't do that yet.